client/extensions/woocommerce/app/promotions/promotion-form-discount-type-and-amount.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+import { find, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSelect from 'components/forms/form-select';
+import FormTextInputWithAffixes from 'components/forms/form-text-input-with-affixes';
+import PriceInput from 'woocommerce/components/price-input';
+
+function renderPercentInput( amount, onAmountChange ) {
+	// TODO: Consider making a FormPercentInput general-purpose component?
+	return (
+		<FormTextInputWithAffixes
+			type="number"
+			min="0"
+			max="100"
+			suffix="%"
+			placeholder="0"
+			value={ amount }
+			onChange={ onAmountChange }
+		/>
+	);
+}
+
+function renderPriceInput( amount, onAmountChange, currency ) {
+	return (
+		<PriceInput
+			currency={ currency }
+			value={ amount }
+			onChange={ onAmountChange }
+		/>
+	);
+}
+
+const inputRenderers = {
+	percent: renderPercentInput,
+	price: renderPriceInput,
+};
+
+const inputValidators = {
+	percent: ( amount ) => ( amount >= 0 && amount <= 100 ),
+	price: ( amount ) => ( amount >= 0 ),
+};
+
+function renderDiscountTypeOption( { value, text } ) {
+	return ( <option key={ value } value={ value }>{ text }</option> );
+}
+
+const PromotionFormDiscountTypeAndAmount = ( {
+	discountTypesAvailable,
+	discountTypeValue,
+	amount,
+	currency,
+	onDiscountTypeSelect,
+	onAmountChange,
+	translate,
+} ) => {
+	const discountType = find( discountTypesAvailable, ( d ) => ( discountTypeValue === d.value ) );
```

Comment:
Optional. lodash find is easier to grok at a glance

Reply:
This is lodash `find` -- see import above.

Reply:
Oh - there's a simpler syntax that available - e.g. `find( discountTypesAvailable, { value: discountTypeValue } )`

But I won't hold up this review for that

Reply:
Oh, I always forget about that syntax for `find`. I'll try to remember that.